src/components/ChatUI.tsx

```diff
@@ -458,6 +480,12 @@ const ChatUI: React.FC<ChatUIProps & { embedded?: boolean }> = ({
     initiateChatBackend();
   }, [open]);
 
+  useEffect(() => {
+    return () => {
+      mcpIntegration.cleanup();
+    };
+  }, []);
```

Comment:
I don't think this is doing anything if we return the function without calling it